service/kernel/verbose_mode.cpp

```diff
@@ -178,14 +172,21 @@ void json::write_escape(const char * const str)
     write('"');
 }
 
+void json::write_key(const char * const key)
+{
+    write('"');
+    write(key);
+    write("\":");
+}
+
 json::~json()
 {
     finalize();
 }
 
 void json::write(const char * const str)
 {
-    if (0 > puts(str)) exit(1);
+    if (0 > printf("%s", str)) exit(1);
```

Comment:
exit?

Reply:
averbukh told that we mush handle functions return value

Reply:
I think exit() Is the worst way to do it from customer point of view)
Also, I think SDL doesn't allow this

Reply:
> I think exit() Is the worst way to do it from customer point of view)
Not quite sure. If someone forced library to work in verbose mode and it can't write anything it will be quite handy to terminate here

> Also, I think SDL doesn't allow this
Why. We shutting down application if we met unknown state

Reply:
I suggest that killing the application on the library side is completely bad idea.
Why not use simple error handling in this case?

Reply:
Also, verbosity is support functionality, not primal thing. If it doesn't work - it's not a critical issue.
I'm sure that the users will be not happy if their application is killed without any notifications what happened.